### ISSUE TITLE
Add look and feel to space/create

### ISSUE THREAD

**Issue by caseydriscoll** [new-day-international/reddit#218]
This is a mess.

Simply, we'd like a user to be able to upload a space's photo on the create space form. Right now they can only add the photo on the edit space form, a procedure that takes several steps to get to, if they even know how and have the will.

The ultimate problem is that the reddit code needs a <Subreddit> to attach the image to. Right now that happens only after the space is created after submission of the create form.

Here is the irc conversation that explains the rest:

```
09:30 caseydriscoll: last checkbox, want to add "look and feel" image upload to /spaces/create
09:31 caseydriscoll: right now you have to 'edit' the space after creation to add an image
09:31 caseydriscoll: createsubreddit.html line 435
09:31 caseydriscoll: these fuckers
09:31 caseydriscoll: it is a form template
09:32 caseydriscoll: that both edit and create space use
09:34 caseydriscoll: but at 435 there is an if condition, that for some bizarre reason asks that a "site" already exist (ie only the edit form my pass)
09:35 caseydriscoll: because the upload form uses the 'sites' parameters (I count three like     thing.site._fullname)
09:35 caseydriscoll: in some hidden forms
09:36 caseydriscoll: this is all a load of bs, i have no idea what the design theory was
09:36 caseydriscoll: in rails you would instantiate a 'new' blank object
09:36 caseydriscoll: so that it would have a 'site' to assign to
09:37 caseydriscoll: for example, here is "thing" in the edit and create forms respectively
09:37 caseydriscoll: {'render_class': <class 'r2.lib.pages.pages.CreateSubreddit'>, 'name': '', 'site': <Subreddit 41>}
09:37 caseydriscoll: {'render_class': <class 'r2.lib.pages.pages.CreateSubreddit'>, 'name': '', 'site': None}
09:37 caseydriscoll: in the create form site is "none" when it should be a newly instantiated subreddit number, in my opinion
09:38 caseydriscoll: not sure how to proceed
09:38 caseydriscoll: just want the look and feel image upload form to show up on the create space form
09:38 myers: caseydriscoll: so you are changing the behavour of the form and changing the way it looks?
09:39 caseydriscoll: the ticket mostly just called for looks
09:39 caseydriscoll: i appended the behavior change
09:39 caseydriscoll: thinking it would be a simple copy/paste to another template
09:40 caseydriscoll: but it doesnt seem like an easy fix to add it to the create form
09:40 caseydriscoll: given the dependency on thing.site
09:40 caseydriscoll: so "yes" to your question
09:40 caseydriscoll: although im hoping to not change the behavior if necessary
09:41 caseydriscoll: just make it show up
09:41 myers: ffunch: that error sentry told us about is related to https://github.com/new-day-international/reddit/issues/212
09:42 ffunch: myers: Yeah, it should no longer exist in test
09:42 myers: caseydriscoll: it could be that the upload image code works as a whole other form.
09:43 ffunch: myers: should I put that one to production? I suppose it is stopping some mail from being sent out.
09:43 caseydriscoll: maybe, but the fucked up thing is that the actual image upload code is in utils.html
09:44 myers: ffunch: apparently just one
09:44 caseydriscoll: image_upload > utils.html:304
09:44 caseydriscoll: i think i could disect it, but dont really want to travel down that path without you looking at it
09:44 caseydriscoll: seems too foundational
09:45 myers: caseydriscoll: I was thinking the python code to upload to s3 was going to be there.  It's just the js.  Yeah, not how I would have done it
09:46 myers: I think test driven development would make you want to put that in it's own file so you could test it easy
09:49 caseydriscoll: i have no idea how to do this at the moment, but if we could pass in a blank <Subreddit> to space/create, i think the problem would fix itself?
09:49 caseydriscoll: ???
09:49 caseydriscoll: technical deeeeeeeeeeeebt
09:50 caseydriscoll: the problem being im not sure how the image uploader uses the thing.site.parameters
09:50 caseydriscoll: <input type="hidden" name="sr" value="${thing.site._fullname}"/>
09:50 caseydriscoll: <input type="text" name="header-title" id="header-title"
09:50 caseydriscoll:                 value="${thing.site.header_title}"
09:50 caseydriscoll:                 />
09:50 caseydriscoll: <%utils:image_upload post_target="/api/upload_sr_img"
09:50 caseydriscoll:                               current_image="${thing.site.header}"
09:50 caseydriscoll:                               label="${_('upload header image')}"
09:50 caseydriscoll:                               ask_type="${True}">
09:51 caseydriscoll: <button id="delete-img" class="delete-img btn btn-xs btn-danger"
09:51 caseydriscoll:                    %if not thing.site.header:
09:51 caseydriscoll:                       style="display: none;"
09:51 caseydriscoll:                    %endif
09:51 caseydriscoll:                    onclick="return post_form(this.form, 'delete_sr_header');">
09:51 caseydriscoll: those are all 4 cases i believe
09:52 caseydriscoll: in the first instance, it needs to know the _fullname for assignment/recall later?
09:52 caseydriscoll: which is the effed up thing
09:52 caseydriscoll: cause it should be by site id or something instead of name, imho
09:53 caseydriscoll: at any rate, unless can be fixed with reasonable speed, id like to take the spec that i appended out of the ticket and move it up the 3
09:54 caseydriscoll: it was originally just cosmetic changes which are finished
09:57 myers: caseydriscoll: I don't think it's going to be an easy fix
09:57 caseydriscoll: ill make a new ticket with that spec
09:57 caseydriscoll: and promote the other one
09:57 myers: because it uploads the image right away
09:58 myers: and expects to have a created subreddit object to attach it to
```
